src/python/WMComponent/AgentStatusWatcher/ResourceControlUpdater.py

```diff
@@ -333,3 +301,44 @@ def updateSiteState(self, siteName, state):
             logging.error(str(ex))
             logging.error("Traceback: \n%s", traceback.format_exc())
         return
+
+    def checkTaskSlotsChanges(self, siteName, CPUBound, IOBound):
+        """
+        _checkTaskSlotsChanges_
+
+        Update the CPU and IOBound slots for a given site.
+        """
+        siteTaskSlots = self.resourceControl.thresholdBySite(siteName)
+        taskCPUPending = max(int(CPUBound / self.agentsNumByTeam * self.pendingSlotsTaskPercent / 100), self.minCPUSlots)
+        taskIOPending = max(int(IOBound / self.agentsNumByTeam * self.pendingSlotsTaskPercent / 100), self.minIOSlots)
+
+        updateTasks = False
+        if siteTaskSlots[0]['task_type'] in self.tasksCPU and siteTaskSlots[0]['task_pending_slots'] != taskCPUPending:
+            updateTasks = True
+        elif siteTaskSlots[0]['task_type'] in self.tasksIO and siteTaskSlots[0]['task_pending_slots'] != taskIOPending:
+            updateTasks = True
+
+        if siteName == "T2_IT_Rome":
+            logging.info("AMR Rome taskCPUPending %s and taskIOPending %s", taskCPUPending, taskIOPending)
+            logging.info("AMR Rome siteTaskSlots[0] %s", siteTaskSlots[0])
+
+        if updateTasks:
+            logging.debug("Updating %s CPU tasks thresholds for pend/runn: %d/%d", siteName,
+                          taskCPUPending, CPUBound)
+            self.resourceControl.insertThreshold(siteName, taskType=self.tasksCPU, maxSlots=CPUBound,
+                                                 pendingSlots=taskCPUPending)
+            logging.debug("Updating %s IO tasks thresholds for pend/runn: %d/%d", siteName,
+                          taskIOPending, IOBound)
+            self.resourceControl.insertThreshold(siteName, taskType=self.tasksIO, maxSlots=IOBound,
```

Comment:
It seems this is not really taking effect, because I see IOBound slots getting updated every cycle.
I just can't find any flaw anywhere and I also failed to get logging records out of the ResourceControl and DAOs :-(